PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:16 + add:0, change matches:
-    _inherit = ["account.move"]
-    _inherit = ["account.move.line"]
-    _inherit = ["account.payment"]
-    _inherit = ["account.tax"]
-    _inherit = ['account.analytic.applicability']
-    _inherit = ['account.analytic.account']
-    _inherit = ['hr.department']
-    _inherit = ['hr.employee']
-    _inherit = ['hr.employee.public']
-    _inherit = ['res.users']
-    _inherit = ['ir.actions.report']
-    _inherit = ['ir.attachment']
-    _inherit = ["product.product"]
-    _inherit = ["product.template"]
-    _inherit = ["res.company"]
-    _inherit = ['res.config.settings']

Total Changes: 35

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_expense/models/account_move.py
+++ b/addons/hr_expense/models/account_move.py
@@ -7,7 +7,7 @@ from odoo.tools.misc import frozendict
 
 
 class AccountMove(models.Model):
-    _inherit = ["account.move"]
+    _inherit = "account.move"
 
     expense_sheet_id = fields.Many2one(comodel_name='hr.expense.sheet', ondelete='set null', copy=False, index='btree_not_null')
     show_commercial_partner_warning = fields.Boolean(compute='_compute_show_commercial_partner_warning')

--- a/addons/hr_expense/models/account_move_line.py
+++ b/addons/hr_expense/models/account_move_line.py
@@ -7,7 +7,7 @@ from odoo.tools.misc import frozendict
 
 
 class AccountMoveLine(models.Model):
-    _inherit = ["account.move.line"]
+    _inherit = "account.move.line"
 
     expense_id = fields.Many2one('hr.expense', string='Expense', copy=True) # copy=True, else we don't know price is tax incl.
 

--- a/addons/hr_expense/models/account_payment.py
+++ b/addons/hr_expense/models/account_payment.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class AccountPayment(models.Model):
-    _inherit = ["account.payment"]
+    _inherit = "account.payment"
 
     expense_sheet_id = fields.Many2one(related='move_id.expense_sheet_id')
 

--- a/addons/hr_expense/models/account_tax.py
+++ b/addons/hr_expense/models/account_tax.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class AccountTax(models.Model):
-    _inherit = ["account.tax"]
+    _inherit = "account.tax"
 
     def _hook_compute_is_used(self, taxes_to_compute):
         # OVERRIDE in order to fetch taxes used in expenses

--- a/addons/hr_expense/models/analytic.py
+++ b/addons/hr_expense/models/analytic.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class AccountAnalyticApplicability(models.Model):
-    _inherit = ['account.analytic.applicability']
+    _inherit = 'account.analytic.applicability'
     _description = "Analytic Plan's Applicabilities"
 
     business_domain = fields.Selection(
@@ -25,7 +25,7 @@ class AccountAnalyticApplicability(models.Model):
 
 
 class AccountAnalyticAccount(models.Model):
-    _inherit = ['account.analytic.account']
+    _inherit = 'account.analytic.account'
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_account_in_analytic_distribution(self):

--- a/addons/hr_expense/models/hr_department.py
+++ b/addons/hr_expense/models/hr_department.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class HrDepartment(models.Model):
-    _inherit = ['hr.department']
+    _inherit = 'hr.department'
 
     def _compute_expense_sheets_to_approve(self):
         expense_sheet_data = self.env['hr.expense.sheet']._read_group([('department_id', 'in', self.ids), ('state', '=', 'submit')], ['department_id'], ['__count'])

--- a/addons/hr_expense/models/hr_employee.py
+++ b/addons/hr_expense/models/hr_employee.py
@@ -4,7 +4,7 @@ from odoo import fields, models, api
 
 
 class HrEmployeeBase(models.AbstractModel):
-    _inherit = ['hr.employee.base']
+    _inherit = 'hr.employee.base'
 
     filter_for_expense = fields.Boolean(store=False, search='_search_filter_for_expense', groups="hr.group_hr_user")
 
@@ -31,7 +31,7 @@ class HrEmployeeBase(models.AbstractModel):
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     def _group_hr_expense_user_domain(self):
         # We return the domain only if the group exists for the following reason:
@@ -68,13 +68,13 @@ class HrEmployee(models.Model):
 
 
 class HrEmployeePublic(models.Model):
-    _inherit = ['hr.employee.public']
+    _inherit = 'hr.employee.public'
 
     expense_manager_id = fields.Many2one('res.users', readonly=True)
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     expense_manager_id = fields.Many2one(related='employee_id.expense_manager_id', readonly=False)
 

--- a/addons/hr_expense/models/hr_expense.py
+++ b/addons/hr_expense/models/hr_expense.py
@@ -10,6 +10,7 @@ from odoo.tools import email_normalize, float_repr, float_round, is_html_empty
 
 
 class HrExpense(models.Model):
+    _name = 'hr.expense'
     _inherit = ['mail.thread.main.attachment', 'mail.activity.mixin', 'analytic.mixin']
     _description = "Expense"
     _order = "date desc, id desc"

--- a/addons/hr_expense/models/ir_actions_report.py
+++ b/addons/hr_expense/models/ir_actions_report.py
@@ -5,7 +5,7 @@ from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
 
 
 class IrActionsReport(models.Model):
-    _inherit = ['ir.actions.report']
+    _inherit = 'ir.actions.report'
 
     def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
         # OVERRIDE

--- a/addons/hr_expense/models/ir_attachment.py
+++ b/addons/hr_expense/models/ir_attachment.py
@@ -2,7 +2,7 @@ from odoo import models, api
 
 
 class IrAttachment(models.Model):
-    _inherit = ['ir.attachment']
+    _inherit = 'ir.attachment'
 
     @api.model_create_multi
     def create(self, vals_list):

--- a/addons/hr_expense/models/product_product.py
+++ b/addons/hr_expense/models/product_product.py
@@ -2,7 +2,7 @@ from odoo import api, fields, models, _
 
 
 class ProductProduct(models.Model):
-    _inherit = ["product.product"]
+    _inherit = "product.product"
 
     standard_price_update_warning = fields.Char(compute="_compute_standard_price_update_warning")
 

--- a/addons/hr_expense/models/product_template.py
+++ b/addons/hr_expense/models/product_template.py
@@ -6,7 +6,7 @@ from odoo.tools.sql import column_exists, create_column
 
 
 class ProductTemplate(models.Model):
-    _inherit = ["product.template"]
+    _inherit = "product.template"
 
     @api.model
     def default_get(self, fields):

--- a/addons/hr_expense/models/res_company.py
+++ b/addons/hr_expense/models/res_company.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ["res.company"]
+    _inherit = "res.company"
 
     expense_journal_id = fields.Many2one(
         "account.journal",

--- a/addons/hr_expense/models/res_config_settings.py
+++ b/addons/hr_expense/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     hr_expense_alias_prefix = fields.Char(
         'Default Alias Name for Expenses',
